lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread_mutex.c

```diff
@@ -297,6 +297,10 @@ int pthread_mutex_unlock( pthread_mutex_t * mutex )
 
     if( iStatus == 0 )
     {
+        /* Enter the critical section so that
+         * mutex is unlocked AND owner is updated atomically */
+        taskENTER_CRITICAL();
+
         /* Call the correct FreeRTOS mutex unlock function based on mutex type. */
         if( pxMutex->xAttr.iType == PTHREAD_MUTEX_RECURSIVE )
         {
```

Comment:
> FreeRTOS API functions must not be called from within a critical section. 

See: https://www.freertos.org/taskENTER_CRITICAL_taskEXIT_CRITICAL.html

Reply:
Just move the critical section around the update of xTaskOwner.